pkg/virt-launcher/virtwrap/network/podnic.go

```diff
@@ -310,6 +310,39 @@ func (l *podNIC) storeCachedDomainIface(domainIface api.Interface) error {
 	return l.cacheFactory.CacheDomainInterfaceForPID(getPIDString(l.launcherPID)).Write(l.vmiSpecIface.Name, &domainIface)
 }
 
+func (l *podNIC) setState(state cache.PodIfaceState) error {
+	podIfaceCaches := l.cacheFactory.CacheForVMI(l.vmi)
+	podIfaceCache, err := podIfaceCaches.Read(l.vmiSpecIface.Name)
+	if err != nil && !os.IsNotExist(err) {
+		log.Log.Reason(err).Errorf("failed to read pod interface network state from cache, %s", err.Error())
+		return err
+	}
+	if os.IsNotExist(err) {
+		podIfaceCache = &cache.PodCacheInterface{}
+	}
+	podIfaceCache.State = state
+	err = podIfaceCaches.Write(l.vmiSpecIface.Name, podIfaceCache)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to write pod interface network state to cache, %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+func (l *podNIC) state() (cache.PodIfaceState, error) {
```

Comment:
This is the pod nic state and it has three possible states

- networking unprepared
- networking prepared
- networking unknown

one thing I can do is to name state to networkingState and have the following state

- unprepared
- prepared
- unknown

And then add some helper methods IsNetworkingUnprepared(), IsNetworkingPrepared() and IsNetworkingUnknown()

What do you think ?

Reply:
I meant that the casual reader of the code will read "state" and not understand what this is.

Is there something more specific we can name this ? 

I guess that if you name the auxiliary methods well it does not matter:
  - hasStartedNetworkingConfiguration
  - isFinishedNetworkingConfiguration
  - ... 

Not sure.

Reply:
It's a matter to use correct names for `PodIfaceState` what do you think of:

- networkingPreparationPending
- networkingPreparationStarted
- networkingPreparationFinished

I like to use the "prepare" word since the infra configurator use "prepare" too.

Reply:
Oh, now I see the motivation - short-sighted of me not seeing it. It is indeed a good point. 

OK, I guess we should document this afterwards as part of the overhaul of the network design doc (which currently holds no truth).

I would use `network` instead of `networking`, but I guess either work.

Reply:
ack, renaming states, thanks!